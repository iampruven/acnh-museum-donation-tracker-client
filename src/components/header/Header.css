.acnh-title {
  font-family: Cabin Sketch;
  background-color: white;
  border: 5px gold solid;
  border-radius: 20px;
  text-align: center;
  font-size:32px;
  width:50%;
}
.center {
  display: flex;
  justify-content: center;
}
.color-font{
    color:white;
    font-weight: 900;
}
nav {
  background-color: #517664;
}

a {
  color: black;
  cursor: pointer;
  display: flex;
  align-items: space-around;
  text-decoration: none;
  font-weight: 900;
}
a:hover, a:focus, a:active {
  color:gold;
}

ul {
  list-style-type: none;
  padding: 0;
  margin: 0;
  text-align: center;
}

nav ul li {
  display: inline-block;
  text-decoration: none;
  font-size: 20px;
  padding: 10px;
  text-align: center;
}

@media only screen and (max-width: 600px) {
    .direct {
        display:flex;
        flex-direction: column;
    }
  }