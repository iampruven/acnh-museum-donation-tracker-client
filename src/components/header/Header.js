import React from "react";
import { Link } from "react-router-dom";
import TokenService from "../../services/token-service";
import "./Header.css";
export default class Header extends React.Component {

  
  handleLogoutClick = () => {
    TokenService.clearAuthToken()
    this.props.history.go('/login')
  };

  renderLogoutLink() {
    return (
      <div>
        <Link onClick={this.handleLogoutClick} to="/" className="color-font">
          Logout
        </Link>
      </div>
    );
  }
  renderLoginLink() {
    return (
      <div>
        <Link to="/login" className="color-font">Login</Link>
      </div>
    );
  }
  renderAddItemLink(){
    return(
      <div>
<Link to="/addItem" className="color-font">Add New Item</Link>
      </div>
    )
  }
  renderRegisterLink(){
    return(
      <div>
<Link to="/register" className="color-font">Register</Link>
      </div>
    )
  }
  render() {

    return (
      <section>
        <nav>
          <ul className="direct">
          <li><Link to="/" className="color-font">Home</Link></li>
          <li>{TokenService.hasAuthToken() ? this.renderAddItemLink(): ''}</li>
          <li>{TokenService.hasAuthToken()
            ? this.renderLogoutLink()
            : this.renderLoginLink()}</li>
          <li>{TokenService.hasAuthToken()? '': this.renderRegisterLink() } </li>
          
           
           </ul>
        </nav>
        <div className="center">
          <h1 className="acnh-title">ACNH: Museum Donation Tracker</h1>
        </div>
      </section>
    );
  }
}
