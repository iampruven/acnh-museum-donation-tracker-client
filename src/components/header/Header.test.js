import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
//smoke test
describe('renders without crashing',()=>{
    it('it renders without crashing',()=>{
        const div = document.createElement('div');

        ReactDOM.render(<MemoryRouter><Header /></MemoryRouter>,div);
        ReactDOM.unmountComponentAtNode(div);
    })


//snapshot test
    it('renders the UI as expected', ()=>{
        const tree = renderer
            .create(<MemoryRouter><Header /></MemoryRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })



})