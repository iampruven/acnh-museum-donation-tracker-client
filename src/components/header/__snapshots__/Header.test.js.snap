// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders the UI as expected 1`] = `
<section>
  <nav>
    <ul
      className="direct"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        
      </li>
      <li>
        <div>
          <a
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </li>
      <li>
        <div>
          <a
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </div>
         
      </li>
    </ul>
  </nav>
  <div>
    <h1
      className="acnh-title center"
    >
      ACNH: Museum Donation Tracker
    </h1>
  </div>
</section>
`;
